cmd/dockerd/daemon.go

```diff
@@ -287,24 +239,39 @@ func (cli *DaemonCli) start() (err error) {
 		"graphdriver": d.GraphDriverName(),
 	}).Info("Docker daemon")
 
-	cli.initMiddlewares(api, serverConfig)
-	initRouter(api, d, c)
+	// setup the default routes
+	defaultRoutes := getRouters(d, c)
+	api.InitRouter(utils.IsDebugEnabled(), defaultRoutes...)
+
+	// setup the introspections routes without cluster capabilities
```

Comment:
I don't really like making an exception here. Only reason I can see for not allowing swarm methods here is that user can abuse is or it is not very useful as user can't control if they are on a manager or not. Both or these don't seem valid as other methods are very dangerous as well and this isn't suitable for many other use-cases as well atm because user needs to fully trust the container. We should add more complicated ways of limiting this api in the future(as discussed) but if this remains as fully privileged way to access daemon in global context I don't see why we should forbid `swarm init/join/leave`.